ms2query/app_helpers.py

```diff
@@ -0,0 +1,303 @@
+            testspectrum are not implemented yet, so network plotting will not
+            work for this library.</span></p>""", unsafe_allow_html=True)
+
+    return library_spectrums, test_sim_matrix
+
+
+def get_model() -> Tuple[Union[Word2Vec, None], Union[int, None]]:
+    """Return (Word2Vec model, model number) and print some info in the app
+    """
+    model_file = st.sidebar.text_input(
+        "Enter filename of Spec2Vec model (with path):")
+    st.write("#### Spec2Vec model")
+    model = None
+    model_num = None
+    if model_file:
+        if model_file.endswith(".model"):
+            st.write("Your selected model:", os.path.split(model_file)[-1])
+            model = Word2Vec.load(model_file)
+            # todo: change this when multiple models are added for caching
+            model_num = 0
+        else:
+            st.write("""<p><span style="color:red">Model file extension should
+            be .model, please try again.</span></p>""", unsafe_allow_html=True)
+    return model, model_num
+
+
+def do_spectrum_processing(query_spectrums: List[Spectrum],
+                           library_spectrums: List[Spectrum]) -> Tuple[
+                           List[SpectrumDocument], List[SpectrumDocument]]:
+    """Process query, library into SpectrumDocuments and write processing info
+
+    Args:
+    -------
+    query_spectrums:
+        Query spectra in matchms.Spectrum format
+    library_spectrums:
+        Library spectra in matchms.Spectrum format
+    """
+    st.write("## Post-process spectra")
+    st.write("""Spec2Vec similarity scores rely on creating a document vector
+    for each spectrum. For the underlying word2vec model we want the documents
+    (=spectra) to be more homogeneous in their number of unique words. Assuming
+    that larger compounds will on average break down into a higher number of
+    meaningful fragment peaks we reduce the document size of each spectrum
+    according to its parent mass.
+    """)
+    # todo: we could add some buttons later to make this adjustable
+    settings = set_spec2vec_defaults()
+    with st.beta_expander("View processing defaults"):
+        st.markdown(
+            f"""* normalize peaks (maximum intensity to 1)\n* remove peaks 
+            outside [{settings["mz_from"]}, {settings["mz_to"]}] m/z window
+            \n* remove spectra with < {settings["n_required"]} peaks\n* reduce
+            number of peaks to maximum of {settings["ratio_desired"]} * parent
+            mass\n* remove peaks with intensities <
+            {settings["intensity_from"]} of maximum intensity (unless this
+            brings number of peaks to less than 10)\n* add losses between m/z
+            value of [{settings["loss_mz_from"]}, {settings["loss_mz_to"]}]""")
+
+    documents_query = process_spectrums(query_spectrums, **settings)
+    documents_library = process_spectrums(library_spectrums, **settings)
```

Comment:
Just a comment:
If we later set a default library, we could skip this step and import already processed data.

Reply:
Yes i was thinking the same. But for now let's keep it for testing